apps/cowswap-frontend/src/modules/onchainTransactions/updaters/FinalizeTxUpdater/services/checkOnChainTransaction.ts

```diff
@@ -10,19 +10,26 @@ export function checkOnChainTransaction(transaction: EnhancedTransactionDetails,
   const { chainId, transactionsCount, dispatch, getReceipt, lastBlockNumber } = params
   const { hash } = transaction
 
+  // Update the last checked blockNumber
+  const setTxLastBlockNumber = () => {
+    dispatch(checkedTransaction({ chainId, hash, blockNumber: lastBlockNumber }))
+  }
+
   // Get receipt for transaction, and finalize if needed
   const { promise, cancel } = getReceipt(hash)
+
   promise
     .then((receipt) => {
       if (receipt) {
         // If the tx is mined. We finalize it!
         finalizeEthereumTransaction(receipt, transaction, params)
       } else {
-        // Update the last checked blockNumber
-        dispatch(checkedTransaction({ chainId, hash, blockNumber: lastBlockNumber }))
+        setTxLastBlockNumber()
       }
     })
     .catch((error) => {
+      setTxLastBlockNumber()
```

Comment:
One of ways where we could stuck into Pending... state.
We always must add `lastBlockNumber` to the transaction.